### PR TITLE
Added class/libtype for collections and library search function

### PR DESCRIPTION
**Pull request by DirtyCajunRice**
Allows for a functioning collections object.

### CODE REVIEW

**Comment by Hellowlol** [comment]
Thanks for the pr! Great job. I have some questions and some feedback. 

Can you make sure the attribute type is correct (use the utils.cast method) the xml is all strings int should int, date should be date time etc. 

I’m not sure if it should be called collection or collections. I think the latter, I think we are collections on the media items. 

How does this differ from our current solution? 
Is your collection compatible with add remote collections or is this handled another way by plex now?

**Comment by DirtyCajunRice** [comment]
1. ok i will check out utils.cast
2. I would LIKE to get rid of the old collection completely but until i figure out how to have it parse the prior to get the latter i left it
3. currently collection is just a tag; it doesnt show you ANY information about the MediaType that now exists or let you manipulate it
4. add remote collections?

**Comment by Hellowlol** [comment]
You can add or remove collections on the plexapi movies episodes etc. see https://github.com/pkkid/python-plexapi/blob/master/plexapi/base.py#L358

**Comment by DirtyCajunRice** [comment]
Ah. No. So thats where its plex not tying things together. Collections still require the tag, and the Collection object shows the tag thats directly tied to it, but the tag does not show the collection it creates. Its a downward waterfall. 

The Collection tag requires:
```
TAG = 'Collection'
FILTER = 'collection'
```
Whereas the Collection View requires:
```
TAG = 'Directory'
TYPE = 'collection'
```

There is a SINGLE way to have the MediaObject.collection show up as the newly created collection object, and that would be to pass the collection tag as a title= iterative over each of them.

Do you want me to implement that?

**Comment by DirtyCajunRice** [comment]
Ah yeah that would be confusing flying blind. Ok. So. Looking at a response from a video gives what you see in your version today still Giving nothing but the collection "Tag" (Title):
```xml
-<Video title="28 Days Later" thumb="/library/metadata/341/thumb/1521153808" art="/library/metadata/341/art/1521153808" key="/library/metadata/341" chapterSource="media" updatedAt="1521153808" addedAt="1519064950" originallyAvailableAt="2003-06-27" duration="6787614" tagline="His fear began when he woke up alone. His terror began when he realised he wasn't." year="2003" rating="7.1" summary="Twenty-eight days after a killer virus was accidentally unleashed from a British research facility, a small group of London survivors are caught in a desperate struggle to protect themselves from the infected. Carried by animals and humans, the virus turns those it infects into homicidal maniacs -- and it's absolutely impossible to contain." contentRating="R" type="movie" studio="DNA Films" ratingKey="341">

-<Media duration="6787614" videoProfile="high" videoFrameRate="24p" container="mkv" videoCodec="h264" audioCodec="dca-ma" audioChannels="6" aspectRatio="1.78" height="1080" width="1920" bitrate="39789" id="271" videoResolution="1080" audioProfile="ma">
<Part size="33759242211" key="/library/parts/23596/1519064950/file.mkv" duration="6787614" videoProfile="high" container="mkv" id="23596" file="/movies/28.Days.Later.(2002)/28.Days.Later.(2002).[Remux-1080p].mkv" audioProfile="ma"/>
</Media>

<Genre tag="Horror"/>
<Genre tag="Thriller"/>
<Director tag="Danny Boyle"/>
<Writer tag="Alex Garland"/>
<Country tag="United Kingdom"/>
<Collection tag="28 Days/Weeks Later"/>
<Role tag="Cillian Murphy"/>
<Role tag="Naomie Harris"/>
<Role tag="Noah Huntley"/>
</Video>
```
Looking at a section by collections gives you output of collections that look like this:
```xml
<Directory title="28 Days/Weeks Later" type="collection" key="/library/metadata/937/children" thumb="/library/metadata/937/composite/1519612692?width=400&height=600" updatedAt="1519612692" addedAt="1519612683" summary="" ratingKey="937" minYear="2003" maxYear="2007" childCount="2" index="1" subtype="movie"> </Directory>
```
so when you call on a mediaobj.collections you currently get a Collection object (MediaTag object) with no useful information:
```py
movie = movies.get('28 Days Later')
col = movie.collections[0]
col
<Collection:11339:28-Days/Weeks-Later>
 col.__dict__
{'_server': <PlexServer:http://10.0.10.121:3>, '_data': <Element 'Collection' at 0x7fb9f846de58>, '_initpath': '/library/metadata/341?checkFiles=1&includeExtras=1&includeRelated=1&includeOnDeck=1&includeChapters=1&includePopularLeaves=1&includeConcerts=1&includePreferences=1', '_details_key': '', 'id': 11339, 'role': None, 'tag': '28 Days/Weeks Later', 'key': None, 'librarySectionID': None, 'librarySectionTitle': None, 'librarySectionType': None, 'tagType': None, 'thumb': None}
```
When you get a Collections object from the new class i created you get a useful collection that shows you all the XML information plex has to give on the collection and the ability to access the child movies from the object:
```py
coll = movies.collection(title="28 Days")[0]
coll
<Collections:937:28-Days/Weeks-Later>
{'_server': <PlexServer:http://10.0.10.121:3>, '_data': <Element 'Directory' at 0x7fb9f7beb138>, '_initpath': '/library/sections/2/all?title=28%20Days&type=18&X-Plex-Container-Size=100&X-Plex-Container-Start=0', '_details_key': '', 'ratingKey': '937', 'key': '/library/metadata/937/children', 'type': 'collection', 'title': '28 Days/Weeks Later', 'subtype': 'movie', 'summary': '', 'index': '1', 'thumb': '/library/metadata/937/composite/1519612692?width=400&height=600', 'addedAt': '1519612683', 'updatedAt': '1519612692', 'childCount': '2', 'maxYear': '2007', 'children': [<Movie:341:28-Days-Later>, <Movie:342:28-Weeks-Later>]}
```

Since the tag still exists and plex did not migrate the MediaTag Collection to become a part of the Directory Collection the only way to associate the 2 to go back "Up the waterfall" is to modify the output of the MediaTag Collection to give you a Collections object (which would give you all the same information and much more). I dont think we can DELETE the old collections object though because the Tag/Filter/Type is different.

File: plexapi/library.py [hunk 1]
```diff
@@ -714,3 +726,26 @@ def _loadData(self, data):
...
+        self.subtype = data.attrib.get('subtype')
+        self.summary = data.attrib.get('summary')
+        self.index = utils.cast(int, data.attrib.get('index'))
+        self.thumb = data.attrib.get('thumb')
+        self.addedAt = utils.toDatetime(data.attrib.get('addedAt'))
+        self.updatedAt = utils.toDatetime(data.attrib.get('updatedAt'))
+        self.childCount = utils.cast(int, data.attrib.get('childCount'))
+        self.minYear = utils.cast(int, data.attrib.get('minYear'))
+        self.maxYear = utils.cast(int, data.attrib.get('maxYear'))
+        self.children = self.fetchItems(self.key)
```
Comment (Hellowlol) [inline plexapi/library.py]: fetchItems does http requests. Change this to a property so it’s only loaded if the user asks for the children.

**Pull request merged**